src/EA.Iws.RequestHandlers/NotificationMovements/BulkUpload/PrenotificationContentConsentValidityRule.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace EA.Iws.RequestHandlers.NotificationMovements.BulkUpload
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Core.Movement.Bulk;
+    using Core.Rules;
+    using Domain.NotificationConsent;
+
+    public class PrenotificationContentConsentValidityRule : IBulkMovementPrenotificationContentRule
+    {
+        private readonly INotificationConsentRepository notificationConsentRepository;
+
+        public PrenotificationContentConsentValidityRule(INotificationConsentRepository notificationConsentRepository)
+        {
+            this.notificationConsentRepository = notificationConsentRepository;
+        }
+
+        public async Task<ContentRuleResult<BulkMovementContentRules>> GetResult(List<PrenotificationMovement> movements, Guid notificationId)
+        {
+            return await Task.Run(() =>
+            {
+                var contentValidityResult = MessageLevel.Success;
+                var contentValidityShipmentNumbers = new List<string>();
+
+                foreach (var shipment in movements)
+                {
+                    if (shipment.ShipmentNumber.HasValue && shipment.ActualDateOfShipment.HasValue)
+                    {
+                        var consent = notificationConsentRepository.GetByNotificationId(notificationId);
```

Comment:
I would move the call to the repo outside of the for loop so that you don't call it every time.

Reply:
Call for consent now moved.